### PR TITLE
CONVENTIONS.md: Add section for TLS configuration

### PR DESCRIPTION
**Pull request by Miciah**
#### `CONVENTIONS.md`: s/well-know/well-known/

* `CONVENTIONS.md` (Metrics): Fix a typo ("well-know").

#### `CONVENTIONS.md`: Add section for TLS configuration

Document that components should use the TLS security profile from the cluster APIServer config.

* `CONVENTIONS.md` (TLS configuration): New section.

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -582,3 +585,21 @@ For guidance about how to implement the requirements in practice, refer to the
...
+
+When a component configures TLS for an internal service that belongs to that
+component, such as the component's metrics endpoint or admission webhook, the
+component should use the ciphers and minimum TLS version that are specified in
+the [TLS security profile](enhancements/kube-apiserver/tls-config.md) in the
+cluster [APIServer
+config](https://github.com/openshift/api/blob/be926bb0d7511bd4e11519f794cd71df60454a31/config/v1/types_apiserver.go#L53-L59).
+
+If a component has a service with TLS settings that must be configurable
+independently from the APIServer's, then if possible, re-use [the TLS security
```
Comment (deads2k) [inline CONVENTIONS.md:599]: let's go with must here

File: CONVENTIONS.md [hunk 2]
```diff
@@ -582,3 +585,21 @@ For guidance about how to implement the requirements in practice, refer to the
...
+#### TLS configuration
+
+When a component configures TLS for an internal service that belongs to that
+component, such as the component's metrics endpoint or admission webhook, the
+component should use the ciphers and minimum TLS version that are specified in
+the [TLS security profile](enhancements/kube-apiserver/tls-config.md) in the
+cluster [APIServer
+config](https://github.com/openshift/api/blob/be926bb0d7511bd4e11519f794cd71df60454a31/config/v1/types_apiserver.go#L53-L59).
+
+If a component has a service with TLS settings that must be configurable
```
Comment (deads2k) [inline CONVENTIONS.md:598]: I'd like to see such exceptions reviewed by staff-eng.  Every additional knob makes it harder for cluster-admins to achieve their goals and this is hard to imagine.

File: CONVENTIONS.md (hunk 1 above)
Comment (Miciah) [reply CONVENTIONS.md]: Fixed in https://github.com/openshift/enhancements/compare/7ce9450fb5368d33a6478ac0c2f80321bb0546ad..a9efc4c754f71ed2080a0dff4c69698121d19073.

File: CONVENTIONS.md (hunk 2 above)
Comment (Miciah) [reply CONVENTIONS.md]: Fixed in https://github.com/openshift/enhancements/compare/7ce9450fb5368d33a6478ac0c2f80321bb0546ad..a9efc4c754f71ed2080a0dff4c69698121d19073.

File: CONVENTIONS.md (hunk 1 above)
Comment (Miciah) [reply CONVENTIONS.md]: Should "the component should use the ciphers and minimum TLS version that are specified in the TLS security profile" also be changed to a "must"?

**Pull request merged**